pandas/tests/groupby/test_apply.py

```diff
@@ -1583,3 +1583,110 @@ def f_4(grp):
     e.loc["Pony"] = np.nan
     e.name = None
     tm.assert_series_equal(result, e)
+
+
+# GH58291
+def test_apply_frame_not_as_index_returns_single_index():
+    df = DataFrame(
+        [
+            ["group_a", 0],
+            ["group_a", 2],
+            ["group_b", 1],
+            ["group_b", 3],
+            ["group_b", 5],
+        ],
+        columns=["group", "value"],
+    )
+    gb = df.groupby("group", as_index=False)[["group", "value"]]
+
+    def up_to_two_rows(df: DataFrame) -> DataFrame:
+        return df.head(2)
+
+    result = gb.apply(up_to_two_rows)
+
+    expected = DataFrame(
+        [["group_a", 0], ["group_a", 2], ["group_b", 1], ["group_b", 3]],
+        columns=["group", "value"],
+        index=range(4),
+    )
+    tm.assert_frame_equal(result, expected)
+
+
+# GH58291
+def test_apply_column_groupby_frame_not_as_index_returns_single_index(df):
```

Comment:
What is this testing that we don't have test coverage for?

Reply:
resolved by removing all tests as they are already tested in other tests